Bluetooth: Mesh: use secure storage in ble mesh (#82319)
zephyrproject-rtos/zephyr: doc/releases/migration-guide-4.1.rst

Flash map
=========

Secure storage
==============

* The :kconfig:option:`CONFIG_SECURE_STORAGE_ITS_STORE_MODULE` symbol no longer implies the

>   :kconfig:option:`CONFIG_FLASH` Kconfig option. Platforms using Secure storage must explicitly enable
>   :kconfig:option:`CONFIG_FLASH` Kconfig option.

Reviewer: ```suggestion
  :kconfig:option:`CONFIG_FLASH` Kconfig option. Platforms using Secure storage must explicitly enable it.
```

* The :kconfig:option:`CONFIG_SECURE_STORAGE_ITS_STORE_IMPLEMENTATION_SETTINGS` symbol no longer implies the
  :kconfig:option:`CONFIG_FLASH_MAP` and :kconfig:option:`CONFIG_NVS` Kconfig options.

>   It either no longer selects :kconfig:option:`CONFIG_SETTINGS` Kconfig option.

Reviewer: ```suggestion
  It no longer selects the :kconfig:option:`CONFIG_SETTINGS` Kconfig option either.
```

>   Platforms using Secure storage must explicitly enable :kconfig:option:`CONFIG_FLASH_MAP`,

Reviewer: Using Secure storage with the settings-based ITS store module.

Author: didn't get this. How the final text should look like?

Reviewer: Actually I think it can be made as simple as the following as we previously refer to `CONFIG_SECURE_STORAGE_ITS_STORE_IMPLEMENTATION_SETTINGS`:
```suggestion
  Platforms using it must now explicitly enable :kconfig:option:`CONFIG_FLASH_MAP`,
```

>   :kconfig:option:`CONFIG_SETTINGS` and either the :kconfig:option:`CONFIG_NVS` or
>   :kconfig:option:`CONFIG_ZMS` Kconfig options.

Reviewer: Let's make it more generic:
```suggestion
  :kconfig:option:`CONFIG_SETTINGS` and the desired storage system (e.g. :kconfig:option:`CONFIG_ZMS`).
```
